src/System.Memory/tests/Performance/Perf.Span.Contains.cs

```diff
@@ -0,0 +1,230 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Memory.Tests
+{
+    // Adpated from Perf.Span.IndexOf.cs
+    public class Perf_Span_Contains
+    {
+        private const int InnerCount = 100_000;
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public void SpanContainsChar(int size)
+        {
+            Span<char> charSpan = new char[size];
+            charSpan[size / 2] = '5';
+
+            bool found = true;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        if (!charSpan.Contains('5'))
+                            found = false;
+                    }
+                }
+            }
+
+            Assert.True(found);
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1)]
+        [InlineData(10)]
+        [InlineData(100)]
+        [InlineData(1000)]
+        public void SpanContainsCharAsBytes(int size)
+        {
+            Span<char> charSpan = new char[size];
+            charSpan[size / 2] = '5';
+            Span<byte> byteSpan = MemoryMarshal.AsBytes(charSpan);
+
+            int index = 0;
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        if (charSpan.Contains('5')) // '5' = 53
```

Comment:
Same here and `StringContainsChar(int size)`.

`found = charSpan.Contains('5');`
There is no point setting index and Asserting it equals what we set it to.

Reply:
These tests are copied directly from `Perf.Span.IndexOf.cs`.

Reply:
I don't believe we're using a tautology. There's some Boolean `or` logic in the loop that is checked at the end of the loop.

Reply:
Errr. I see what you mean, OK I will fix it